neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+                     name=self.name,
+                     description=self.description,
+                     **self.annotations)
+
+
+class EpochProxy(BaseEpochProxy):
+
+    def __init__(self, epochs_table, epoch_name=None, index=None):
+        self._epochs_table = epochs_table
+        if index is not None:
+            self._index = index
+            self.shape = (index.sum(),)
+        else:
+            self._index = slice(None)
+            self.shape = epochs_table.n_rows  # untested, just guessed that n_rows exists
+        self.name = epoch_name
+
+    def load(self, time_slice=None, strict_slicing=True):
+        """
+        *Args*:
+            :time_slice: None or tuple of the time slice expressed with quantities.
+                            None is all of the intervals.
+            :strict_slicing: True by default.
+                Control if an error is raised or not when one of the time_slice members
+                (t_start or t_stop) is outside the real time range of the segment.
+        """
+        start_times = self._epochs_table.start_time[self._index]
+        stop_times = self._epochs_table.stop_time[self._index]
+        durations = stop_times - start_times
+        labels = self._epochs_table.tags[self._index]
+
+        return Epoch(times=start_times * pq.s,
+                     durations=durations * pq.s,
+                     labels=labels,
+                     name=self.name)
+
+
+class SpikeTrainProxy(BaseSpikeTrainProxy):
+
+    def __init__(self,  units_table, id):
+        self._units_table = units_table
+        self.id = id
+        self.units = pq.s
+        t_start, t_stop = units_table.get_unit_obs_intervals(id)[0]
+        self.t_start = t_start * pq.s
+        self.t_stop = t_stop * pq.s
+        self.annotations = {"nwb_group": "acquisition"}
+        try:
+            # NWB files created by Neo store the name as an extra column
+            self.name = units_table._name[id]
+        except AttributeError:
+            self.name = None
+        self.shape = None   # no way to get this without reading the data
+
+    def load(self, time_slice=None, strict_slicing=True):
+        """
+        *Args*:
+            :time_slice: None or tuple of the time slice expressed with quantities.
+                            None is the entire spike train.
+            :strict_slicing: True by default.
+                Control if an error is raised or not when one of the time_slice members
+                (t_start or t_stop) is outside the real time range of the segment.
+        """
+        interval = None
+        if time_slice:
+            interval = (float(t) for t in time_slice)  # convert from quantities
+        spike_times = self._units_table.get_unit_spike_times(self.id, in_interval=interval)
+        return SpikeTrain(
+                    spike_times * self.units,
+                    self.t_stop,
+                    units=self.units,
+                    #sampling_rate=array(1.) * Hz,
```

Comment:
https://pynwb.readthedocs.io/en/stable/pynwb.misc.html?highlight=Units#pynwb.misc.Units
```suggestion
                    waveform_rate,
```

Reply:
added to #1013